### DISCUSSION TITLE
.NET 8 SDK with .NET 7 runtime?

### DISCUSSION THREAD

**Discussion by zivkan** [dotnet/dotnet-docker#4619] [Q&A] [answered]
The .NET SDK can build against any runtime the same, or lower, version of the SDK. For example, .NET 8 can build .NET 7 apps. Newer versions of the .NET SDK have build tool improvements (for example, .NET 8 has better support for cross architecture building via docker)

However, for `dotnet test` to work, the project's runtime needs to be installed. In this example, that's the .NET 7 runtime.

So, how do customers use the .NET 8 SDK in a docker image with the .NET 7 runtime installed?

Background information: https://github.com/NuGet/Home/issues/12227#issuecomment-1549097412

**Comment by mthalman** [comment, marked as answer]
You can accomplish this with a one-liner in your Dockerfile:

```Dockerfile
FROM mcr.microsoft.com/dotnet/sdk:8.0-preview

# Import ASP.NET Core runtime and .NET runtime for 7.0
COPY --from=mcr.microsoft.com/dotnet/aspnet:7.0 /usr/share/dotnet/shared /usr/share/dotnet/shared
```

**Reply by richlander** [reply]
Once we push 7.0 Jammy chiseled images, I'd suggest switching to those for those copy, since the `COPY` will be faster.

This is what that address will be: `mcr.microsoft.com/dotnet/aspnet:7.0-jammy-chiseled`

**Comment by turowicz** [comment]
sweet
